### PR TITLE
[expo-font][android] Rewrite android code to Kotlin

### PR DESCRIPTION
**Pull request by kkafar**
# Why

This task was done as a part of migrating expo modules to Kotlin project.

# How

The code had been initially converted from Java to Kotlin with help of Android Studio's conversion tool. Then I was manually adjusted / improved.

# Test Plan

BareExpo (test-suite + ncl)

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).

### CODE REVIEW

File: packages/expo-font/android/src/main/java/expo/modules/font/FontLoaderModule.kt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import expo.modules.interfaces.constants.ConstantsInterface
+
+import java.io.File
+import java.lang.Exception
+
+private const val ASSET_SCHEME = "asset://"
+private const val EXPORTED_NAME = "ExpoFontLoader"
+
+class FontLoaderModule(context: Context) : ExportedModule(context) {
+  private var mModuleRegistry: ModuleRegistry? = null
```
Comment (lukmccall) [inline packages/expo-font/android/src/main/java/expo/modules/font/FontLoaderModule.kt:24]: Can we use ModuleRegistryDelegate instead?

File: packages/expo-font/android/src/main/java/expo/modules/font/FontLoaderModule.kt [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  @ExpoMethod
+  fun loadAsync(fontFamilyName: String, localUri: String, promise: Promise) {
+    try {
+      // TODO: remove Expo references
+      // https://github.com/expo/expo/pull/4652#discussion_r296630843
+      var prefix = ""
+      if (isScoped) {
+        prefix = "ExpoFont-"
+      }
```
Comment (lukmccall) [inline packages/expo-font/android/src/main/java/expo/modules/font/FontLoaderModule.kt:38]:
```suggestion
      val prefix = if (isScoped) {
        prefix = "ExpoFont-"
      } else {
      	""
      }
```

**Review by lukmccall — COMMENTED** [review]
LGTM 💪

**Comment by lukmccall** [comment]
@kkafar, it's ready to be merged?

**Comment by kkafar** [comment]
I believe so!

**Pull request merged**